### PR TITLE
Fetch packages into a folder with their name. Fixes #502.

### LINKED ISSUE #502: Protecting from import leak by fetching package one directory deeper
There were complaints in the NG that DUB does'nt support libraries whose source files are at the top-level directory.

Reference: http://forum.dlang.org/post/ysnfzpvrcvmjdekulfnq@forum.dlang.org
```
- When cloning repositories, dub does not preserve the 
repository's directory name (so e.g. fruit will be cloned to 
~/.dub/fruit-1.0.0/).

Somebody has created a Dub package for my library (excluding 
certain packages, due to point 1a above), and the other day 
someone complained that it doesn't work with Dscanner, because of 
the above issue - the module path "ae.dir.module" does not 
correspond to the filesystem path "ae-1.0.1/dir/module.d".
```

I think this is mostly solvable by fetching dependencies one directory deeper :

Reference: http://forum.dlang.org/post/ofznlktcmxwnpicjbfhv@forum.dlang.org
```
Currently, using ae checkouted in an ae/ directory means having 
an "import leak" meaning that you have to import a path outside 
the checkouted repositery. That's somehow unhygienic like, er, a 
global namespace. But yeah DUB could isolate checkouts one 
directory deeper (ae-1.0.0/ae/) to make this a non-problem, this 
would support libraries like ae/ and make DScanner work.
```

What do you think?

### PR DESCRIPTION
**Pull request by s-ludwig**
This changes the behavior of #712 to be backwards compatible with the old single-layer package structure. It also fixes searching within non-managed paths ("dub add-path") to *not* expect the new layout.

### CODE REVIEW

**Comment by skoppe** [comment]
Looks good to me.

**Comment by s-ludwig** [comment]
Thanks! Practical testing didn't show any issues so far either, will merge.

**Comment by MartinNowak** [comment]
Great, that this is merged now. Would be nice if some of the git submodule people could test this and write a small how-to @Cybershadow, @Hackerpilot.

File: source/dub/packagemanager.d [hunk 1]
```diff
@@ -203,9 +203,19 @@ class PackageManager {
 	bool isManagedPackage(Package pack)
 	const {
 		auto ppath = pack.basePackage.path;
+		return isManagedPath(ppath);
+	}
+
+	/** Determines if a specifc path is within a DUB managed package folder.
+
+		By default, managed folders are "~/.dub/packages" and
+		"/var/lib/dub/packages".
```
Comment (MartinNowak) [inline source/dub/packagemanager.d]: Nice, didn't knew there was a system-wide installation path.
This is useful for caching common dependencies, why is it deprecated?

Comment (skoppe) [inline source/dub/packagemanager.d]: Huh, what gives you the idea it's deprecated?

**Pull request merged**